File: LightningConstructs/Source/Dictionary/Properties.cs
using System.Collections.Generic;

namespace Lightning
{
    public partial class Dictionary<TKey, TValue>
    {
        public System.Int32 Length => FreePointer - ReclaimedCount;

        public TValue this[TKey Key]
        {
            get
            {
                System.Int32 ClaimedPointer = Search(Key);
                if (ClaimedPointer == -1) { throw new System.Collections.Generic.KeyNotFoundException($"Key {Key?.ToString()} was not found in the dictionary."); }
                return BaseValuesArray[ClaimedPointer];
            }
            set
            {
                System.Int32 ClaimedPointer = Search(Key);
                if (ClaimedPointer == -1) { throw new System.Collections.Generic.KeyNotFoundException($"Key {Key?.ToString()} was not found in the dictionary."); }
                    BaseValuesArray[ClaimedPointer] = value;
            }
        } 
    }
}

File: LightningConstructs/Source/Dictionary/Properties.vb
    Public Partial Class Dictionary(Of TKey, TValue)
        Public ReadOnly Property Length As System.Int32
            Get
                Return FreePointer - ReclaimedCount
            End Get
        End Property

        Default Public Property Item(ByVal Key As TKey) As TValue
            Get
                Dim ClaimedPointer As System.Int32 = Search(Key)

                If ClaimedPointer = -1 Then
                    Throw New System.Collections.Generic.KeyNotFoundException($"The key was not found in the dictionary!")
                End If

                Return BaseValuesArray(ClaimedPointer)
            End Get
            Set(ByVal value As TValue)
                Dim ClaimedPointer As System.Int32 = Search(Key)

                If ClaimedPointer = -1 Then
                    Throw New System.Collections.Generic.KeyNotFoundException($"The key was not found in the dictionary!")
                End If

                BaseValuesArray(ClaimedPointer) = value
            End Set
        End Property
    End Class

